JitsiConference.js

```diff
@@ -1623,7 +1786,519 @@ JitsiConference.prototype.broadcastEndpointMessage = function(payload) {
+    }
+};
+
+/**
+ * Sets new P2P status and updates some events/states hijacked from
+ * the <tt>JitsiConference</tt>.
+ * @param {boolean} newStatus the new P2P status value, <tt>true</tt> means that
+ * P2P is now in use, <tt>false</tt> means that the JVB connection is now in use
+ * @private
+ */
+JitsiConference.prototype._setP2PStatus = function(newStatus) {
+    if (this.p2pEstablished === newStatus) {
+        logger.error(`Called _setP2PStatus with the same status: ${newStatus}`);
+
+        return;
+    }
+    this.p2pEstablished = newStatus;
+    if (newStatus) {
+        logger.info('Peer to peer connection established!');
+    } else {
+        logger.info('Peer to peer connection closed!');
+    }
+
+    // Clear dtmfManager, so that it can be recreated with new connection
+    this.dtmfManager = null;
+
+    // Update P2P status
+    this.eventEmitter.emit(
+        JitsiConferenceEvents.P2P_STATUS,
+        this,
+        this.p2pEstablished);
+
+    // Refresh connection interrupted/restored
+    this.eventEmitter.emit(
+        this.isConnectionInterrupted()
+            ? JitsiConferenceEvents.CONNECTION_INTERRUPTED
+            : JitsiConferenceEvents.CONNECTION_RESTORED);
+};
+
+/**
+ * Starts new P2P session.
+ * @param {string} peerJid the JID of the remote participant
+ * @private
+ */
+JitsiConference.prototype._startP2PSession = function(peerJid) {
+    this._maybeClearDeferredStartP2P();
+    if (this.p2pJingleSession) {
+        logger.error('P2P session already started!');
+
+        return;
+    }
+
+    this.isP2PConnectionInterrupted = false;
+    this.p2pJingleSession
+        = this.xmpp.connection.jingle.newP2PJingleSession(
+                this.room.myroomjid,
+                peerJid);
+    this.p2pJingleSession.setSSRCOwnerJid(this.room.myroomjid);
+
+    logger.info('Created new P2P JingleSession', this.room.myroomjid, peerJid);
+
+    this.p2pJingleSession.initialize(true /* initiator */, this.room, this.rtc);
+
+    // NOTE one may consider to start P2P with the local tracks detached,
+    // but no data will be sent until ICE succeeds anyway. And we switch
+    // immediately once the P2P ICE connects.
+    const localTracks = this.getLocalTracks();
+
+    logger.info(`Adding ${localTracks} to P2P...`);
+    this.p2pJingleSession.addLocalTracks(localTracks).then(
+        () => {
+            logger.info(`Added ${localTracks} to P2P`);
+            logger.info('About to send P2P \'session-initiate\'...');
+            this.p2pJingleSession.invite();
+        },
+        error => {
+            logger.error(`Failed to add ${localTracks} to P2P`, error);
+        });
+};
+
+/**
+ * Method when called will decide whether it's the time to start or stop
+ * the P2P session.
+ * @param {boolean} userLeftEvent if <tt>true</tt> it means that the call
+ * originates from the user left event.
+ * @private
+ */
+JitsiConference.prototype._startStopP2PSession = function(userLeftEvent) {
```

Comment:
sure, if you think it doesn't make things more confusing to omit `Session`.  i'm in favor of making variables descriptive over keeping them short